src/types/types.ts

```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
 
 export interface DataSetChangeById extends DataSetPatch, InternalId {}
 
+export type Label = ExternalId;
+
+export type LabelFilter =
+  | LabelContainsFilter
+  | LabelContainsAnyFilter
+  | LabelContainsAllFilter;
+
+export interface LabelContainsFilter {
+  contains: Label;
+}
+
+export interface LabelContainsAnyFilter {
+  containsAny: Label[];
+}
+
+export interface LabelContainsAllFilter {
+  containsAll: Label[];
+}
+
+export interface LabelsPatch {
+  /**
+   * A list of labels to add to the resource
+   */
+  add?: Label[];
+  /**
+   * A list of labels to remove to the resource
+   */
+  remove?: Label[];
+}
+
+export interface ExternalLabelDefinition extends Label {
+  /**
+   * Name of the label.
+   */
+  name: string;
+
+  /**
+   * Description of the label.
+   */
+  description?: string;
+}
+
+export interface LabelDefinition extends ExternalLabelDefinition {
+  createdTime: Date;
+}
+
+export interface LabelDefinitionFilter {
+  filter?: {
+    /**
+     * Returns the label definitions matching that name.
+     */
+    name?: string;
+
+    /**
+     * Filter external ids starting with the prefix specified
+     */
+    externalIdPrefix?: ExternalIdPrefix;
+  };
+}
+
+export interface LabelDefinitionFilterRequest
+  extends LabelDefinitionFilter,
+    Cursor,
+    Limit {}
```

Comment:
instead of `Cursor, Limit`
extends `FilterQuery`

Reply:
maybe make it similar to `DataSetFilterRequest` so we don't empty types?
I mean move `filter?: LabelDefinitionFilter ` in here and unwrap the current `LabelDefinitionFilter`